CMakeLists.txt

```diff
@@ -13,23 +24,53 @@ else ()
 endif ()
 unset(_parent)
 
+include(CTest)
+if (NOT CPack_CMake_INCLUDED)
+  include(CPack)
+endif ()
+include(GNUInstallDirs)
+include(CMakeDependentOption)
+set(CMAKE_INSTALL_CMAKEDIR
+    "${CMAKE_INSTALL_LIBDIR}/cmake"
+    CACHE PATH "Base directory for installing cmake specific artifacts")
+set(INSTALL_VAST_CMAKEDIR "${CMAKE_INSTALL_CMAKEDIR}/vast")
+
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+include(Helpers)
+include(MacDependencyPaths)
+
+# Build configuration
+
 option(VAST_RELOCATABLE_INSTALL "Enable relocatable installations" ON)
 option(ENABLE_ZEEK_TO_VAST "Build zeek-to-vast" ON)
 option(VAST_STATIC_EXECUTABLE "Link VAST statically")
-option(VAST_NO_ARROW "Build without Apache Arrow" OFF)
-set(BUILD_SHARED_LIBS_DEFAULT ON)
-if (VAST_STATIC_EXECUTABLE)
-  set(BUILD_SHARED_LIBS_DEFAULT OFF)
+cmake_dependent_option(
+  BUILD_SHARED_LIBS "Build shared libraries instead of static" ON
+  "NOT VAST_STATIC_EXECUTABLE" OFF)
+cmake_dependent_option(NO_UNIT_TESTS "Don't build unit tests" OFF
+                       "NOT CMAKE_CROSS_COMPILING" ON)
+cmake_dependent_option(VAST_NO_ARROW "Build without Apache Arrow" OFF
+                       "NOT ${CMAKE_SYSTEM_NAME} STREQUAL \"FreeBSD\"" ON)
+
+# Keep make output sane
+set(CMAKE_VERBOSE_MAKEFILE
+    false
+    CACHE STRING "Show all outputs including compiler lines." FORCE)
+
+# Force colored output for the Ninja generator
+option(VAST_COLOR_NINJA "Enable ANSI-colored output for the Ninja generator."
+       ON)
+if ("${CMAKE_GENERATOR}" STREQUAL "Ninja" AND VAST_COLOR_NINJA)
+  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    add_compile_options(-fdiagnostics-color=always)
+  elseif ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+    add_compile_options(-fcolor-diagnostics)
+  endif ()
```

Comment:
Line 67 and 216 both set the `-fdiagnostics-color` for Ninja. The latter should be removed I think.

Reply:
Great catch!